src/cmd/watch.rs

```diff
@@ -87,11 +89,15 @@ fn find_gitignore(book_root: &Path) -> Option<PathBuf> {
 }
 
 fn filter_ignored_files(ignore: Gitignore, paths: &[PathBuf]) -> Vec<PathBuf> {
+    let current_dir = env::current_dir().expect("Unable to determine the current directory");
```

Comment:
I don't think using the current_dir is correct. The ignore builder is built relative to the book root, which may not be the same as the current_dir. Since paths in the ignore file are going to be relative to the book root, using paths relative to the current directory won't match.

Additionally, mdbook can watch directories outside of the book root (such as `extra_watch_dirs`).

Reply:
update in b0a001c